1.0-Upgrade.md

```diff
@@ -0,0 +1,108 @@
+# kubernetes-deploy 0.31.0 -> krane 1.0.0 migration guide
+
+**THIS MIGRATION GUIDE IS A WORK IN PROGRESS. THINGS CAN STILL CHANGE BEFORE 1.0.0**
+
+`kubernetes-deploy` was renamed `krane` when version 1.0.0 was released. Version 1.0.0 introduced new features and [breaking changes](CHANGELOG.md). This guide will help you transition to version 1.0.0 as smoothly as possible.
+
+**TL;DR**:
+* The command-line interface was redesigned; besides the name change, there are breaking changes in several flags.
+* There are breaking changes in the public API (such as the renaming of the `KubernetesDeploy` namespace to `Krane`, and the change in default values for different arguments of the public interface).
+* StatsD metrics will now be generated with the `krane` prefix.
+* `kubernetes-deploy` (now `krane deploy`) / `DeployTask` can no longer deploy global (non-namespaced) resources. A new command called `krane global-deploy` and a related class called `GlobalDeployTask` were added to replace that feature.
+* If you attempt to install two gems that have conflicting executables, `gem install` will warn you but the most recently installed one will win. This means that you can run both `kubernetes-deploy` 0.31.0 and `krane` 1.0.0 side by side by doing:
+
+```bash
+gem install kubernetes-deploy
+gem install krane
+```
+
+## Public API changes
+
+The only breaking change in the public API between version 0.31.0 and 1.0.0 is the renaming of the `KubernetesDeploy` namespace to `Krane`. Besides that, the API of the major public classes (`DeployTask`, `RenderTask`, `RunnerTask`, and `RestartTask`) didn't change at all between 0.31.0 and 1.0.0. If you're curious about this API, check the comment-based docs on these classes or the [rendered documentation at RubyGems.org](https://www.rubydoc.info/gems/kubernetes-deploy/1.0.0/KubernetesDeploy/DeployTask).
```

Comment:
⛏ This sounds smoother: `Otherwise, the APIs ...[keep the same]... have not changed between 0.31.0 and 1.0.0`

I didn't immediately understand what `comment-based docs` were, so maybe something more descriptive  like `check out the comments on the classes themselves` would be more clear? It could also just be me so feel free to ignore.